// Copyright 2023 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package continuous_batching;

// The continuous batcher service definition.
service ContinuousBatcher {
  // Sends a reply containing the actual batch size.
  rpc GetNeedNum (BatchingRequest) returns (BatchingReply) {}
}


// The request message containing the batch size needed by inference engine.
message BatchingRequest {
  int32 need_num = 1;
}

// The response message containing the actual batch size of pending requests 
// in the queue
message BatchingReply {
  int32 actual_batch_size = 1;
}
